evaluation/scripts/service_setup.py

```diff
@@ -0,0 +1,159 @@
+            "deployment": "none-needed-for-openaicom",
+        }
+    return openai_config
+
+def get_search_client():
+    if api_key := os.environ.get("AZURE_SEARCH_KEY"):
+        logger.info("Using Azure Search Service with API Key from AZURE_SEARCH_KEY")
+        azure_credential = AzureKeyCredential(api_key)
+    else:
+        logger.info("Using Azure Search Service with Azure Developer CLI Credential")
+        azure_credential = AzureDeveloperCliCredential()
+
+    return SearchClient(
+        endpoint=f"https://{os.environ['AZURE_SEARCH_SERVICE']}.search.windows.net",
+        index_name=os.environ["AZURE_SEARCH_INDEX"],
+        credential=azure_credential,
+    )
+
+
+def get_openai_client(oai_config: ModelConfiguration):
+    if isinstance(oai_config, AzureOpenAIModelConfiguration):
+        azure_token_provider = None
+        if not os.environ.get("AZURE_OPENAI_KEY"):
+            azure_token_provider = get_bearer_token_provider(
+                AzureDeveloperCliCredential(),
+                "https://cognitiveservices.azure.com/.default",
+            )
+        logger.info(azure_token_provider)
+        return openai.AzureOpenAI(
+            api_version=oai_config.api_version,
+            azure_endpoint=oai_config.azure_endpoint,
+            api_key=oai_config.api_key if os.environ.get("AZURE_OPENAI_KEY") else None,
+            azure_ad_token_provider=azure_token_provider,
+            azure_deployment=oai_config.azure_deployment,
+        )
+    elif isinstance(oai_config, OpenAIModelConfiguration):
+        oai_config: OpenAIModelConfiguration = oai_config
+        return openai.OpenAI(api_key=oai_config.api_key, organization=oai_config.organization)
+    else:
+        raise ValueError(f"Unsupported OpenAI configuration type: {type(oai_config)}")
+
+
+def get_openai_target() -> PromptChatTarget:
+    if os.environ["OPENAI_HOST"] == "azure":
+        logger.info("Using Azure OpenAI Chat Target")
+        deployment = os.environ["AZURE_OPENAI_EVAL_DEPLOYMENT"]
+        endpoint = os.environ["AZURE_OPENAI_EVAL_ENDPOINT"]
+        if api_key := os.environ.get("AZURE_OPENAI_KEY"):
+            return AzureOpenAIChatTarget(
+                deployment_name=deployment,
+                endpoint=endpoint,
+                api_key=api_key,
+            )
+        else:
+            return AzureOpenAIChatTarget(deployment_name=deployment, endpoint=endpoint, use_aad_auth=True)
+    else:
+        logger.info("Using OpenAI Chat Target")
+        return OpenAIChatTarget(api_key=os.environ["OPENAICOM_KEY"])
+
+
+def get_azure_ml_chat_target(
+    chat_message_normalizer: ChatMessageNormalizer = ChatMessageNop,
+) -> AzureMLChatTarget:
+    endpoint = os.environ["AZURE_ML_ENDPOINT"]
+    api_key = os.environ["AZURE_ML_MANAGED_KEY"]
+    return AzureMLChatTarget(
+        endpoint_uri=endpoint,
+        api_key=api_key,
+        chat_message_normalizer=chat_message_normalizer,
+    )
```

Comment:
Does this allow us to talk to models hosted on Hugging Face? Have you tried running this with our latest HF integration feature?